lago/config.py

```diff
@@ -14,88 +14,216 @@
+            except IOError:
+                pass
+        configp.read_dict(get_env_dict(self.root_section))
+        return {s: dict(configp.items(s)) for s in configp.sections()}
+
+    def update_args(self, args):
+        """Update config dictionary with parsed args, as resolved by argparse.
+        Only root positional arguments that already exist will overridden.
+
+        Args:
+            args (namespace): args parsed by argparse
+        """
+
+        for arg in vars(args):
+            if self.get(arg):
+                self._config[self.root_section][arg] = getattr(args, arg)
+
+    def update_parser(self, parser):
+        """Update config dictionary with declared arguments in an argparse.parser
+        New variables will be created, and existing ones overridden.
+
+        Args:
+            parser (argparse.ArgumentParser): parser to read variables from
+
+        """
+
+        self._parser = parser
+        ini_str = argparse_to_ini(parser)
+        configp = configparser.ConfigParser(allow_no_value=True)
+        configp.read_dict(self._config)
+        configp.read_string(ini_str)
+        self._config.update(
+            {
+                s: dict(configp.items(s))
+                for s in configp.sections()
+            }
+        )
+
+    def get(self, *args):
+        """Get a variable from the default section
+        Args:
+            *args (args): dict.get() args
+
+        Returns:
+            str: config variable
+        """
+
+        return self._config[self.root_section].get(*args)
+
+    def __getitem__(self, key):
+        """Get a variable from the default section, good for fail-fast
+        if key does not exists.
+
+        Args:
+            key (str): key
+
+        Returns:
+            str: config variable
+        """
+
+        return self._config[self.root_section][key]
+
+    def get_section(self, *args):
+        """get a section dictionary
+        Args:
+
+        Returns:
+            dict: section config dictionary
+        """
+
+        return self._config.get(*args)
+
+    def get_ini(self, defaults_only=False):
+        """Return the config dictionary in INI format
+        Args:
+            defaults_only (bool): if set, will ignore arguments set by the CLI.
+
+        Returns:
+            str: string of the config file in INI format
+        """
+        if self._parser:
+            if not defaults_only:
+                self._parser.set_defaults(
+                    **self.get_section(self.root_section)
+                )
+            return argparse_to_ini(self._parser)
+        else:
+            configp = configparser.ConfigParser(allow_no_value=True)
+            configp.read_dict(self._config)
+            with StringIO() as out_ini:
+                configp.write(out_ini)
+                return out_ini.getvalue()
+
+    def __repr__(self):
+        return self._config.__repr__()
+
+    def __str__(self):
+        return self._config.__str__()
+
+
+config = ConfigLoad()
```

Comment:
Is this really necessary? Can't it be done on demand instead? (to avoid instantiating it on import, and running it instead on demand)